src/main/java/com/mongodb/hibernate/jdbc/MongoConnection.java

```diff
@@ -94,6 +87,9 @@ public void commit() throws SQLException {
 
     private void doCommit() throws SQLException {
         assertFalse(autoCommit);
+        if (!clientSession.hasActiveTransaction()) {
+            return;
```

Comment:
If we do this, then we should remove the `clientSession.hasActiveTransaction()` check from `setAutoCommit`. However, I am not sure we should do such a change (the same goes about the similar change in `rollback`): `com.mongodb.client.ClientSession.commitTransaction`/`abortTransaction` throw `IllegalStateException` if the `TransactionState` is anything but `IN`. I think, we should preserve that behavior (as it was done before this change), instead of pretending that we have successfully committed / rolled back in situations when we did not commit / roll back anything.

Reply:
This is necessary to work around Hibernate's conceptual transaction mechanism. If we don't do this, the following testing case in `com.mongodb.hibernate.SessionTests` will fail:

```
    @Test
    void testBeginTransaction() {
        assertDoesNotThrow(() -> session.beginTransaction().commit());
    }
```
for the simple reason that physical transaction has not been started. This is a dilemma due to discrepancy between Hibernate conceptual transaction (started by calling `setAutoCommit(false)`) and physical transaction (which is started when first statement is issued).

My biggest concern is if we didn't do this, we might face more trouble during Hibernate integration. Also, I checked existing h2, postgreSQL JDBC driver implementation, they all allow for such behaviour; JDBC spec doesn't disallow it as well.

Reply:
I checked a couple of existing JDBC implementations (including PostgreSQL, H2, SQL Server and MyriaDB). All of them are based on the logic that no-op if no transaction exists.

Reply:
Thank you, I agree.

Let's then

1. either remove the `clientSession.hasActiveTransaction()` check from `setAutoCommit`, and rename the `doCommit` to something like `doCommitIfNeeded`,
2. or add a similar check to `commit`, and replace the `clientSession.hasActiveTransaction()` check in `doCommit` with `assertTrue(clientSession.hasActiveTransaction())`.

Reply:
Good catch again! I adopted your first suggestion.